webbeans-tomcat/src/test/java/org/apache/tomcat/webbeans/test/TomcatNormalScopeProxyFactoryTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.tomcat.webbeans.test;
+
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Writer;
+import java.lang.reflect.Method;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.context.SessionScoped;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.CDI;
+import javax.servlet.ServletRequestEvent;
+
+import org.apache.catalina.Context;
+import org.apache.catalina.connector.Request;
+import org.apache.catalina.connector.Response;
+import org.apache.catalina.core.StandardContext;
+import org.apache.catalina.startup.Tomcat;
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.apache.webbeans.config.WebBeansContext;
+import org.apache.webbeans.spi.ContextsService;
+import org.apache.webbeans.web.tomcat.OpenWebBeansListener;
+import org.junit.Test;
+
+public class TomcatNormalScopeProxyFactoryTest
+{
+    private static final Logger log = Logger.getLogger(TomcatNormalScopeProxyFactoryTest.class.getName());
+
+    @Test
+    public void checkDeserialisation() throws Exception
+    {
+        final File base = dir(new File("target/TomcatNormalScopeProxyFactoryTest-" + System.nanoTime()));
+        final File war = createWar(dir(new File(base, "test")), MyWrapper.class, MySessionScoped.class);
+
+        String sessionId = null;
+        for (final String expected : asList("init", "new"))
+        {
+            final Tomcat tomcat = new Tomcat();
+            tomcat.setPort(0);
+            tomcat.setBaseDir(base.getAbsolutePath());
+            tomcat.getServer().addLifecycleListener(new OpenWebBeansListener());
+
+            final Context ctx = tomcat.addContext("/test", war.getAbsolutePath());
+
+            // needed for Java9
+            if (ctx instanceof StandardContext) {
```

Comment:
from memory it "just" logs a warning or error but does not fail so not important for tests (and surely to handle in tomcat directly?), wdyt?